src/main/configurations/brp-personen-bevragen-soap/Configuration-brp-pb-client-soap.xml

```diff
@@ -85,13 +85,54 @@
 
 			<JsonPipe name="JsonToXml" storeResultInSessionKey="xmlbody" />
 
+			<PutInSessionPipe name="StoreBsn" getInputFromSessionKey="xmlbody">
+				<Param name="bsn" xpathExpression="(//burgerservicenummer)[1]/text()" />
+			</PutInSessionPipe>
+
+			<!-- Check if bsn has a key -->
+			<SenderPipe name="GetPersoonssleutel" storeResultInSessionKey="persoonssleutel">
+				<FixedQuerySender
+					name="FetchPersoonssleutel"
+					queryType="SELECT"
+					query="SELECT * FROM persoonssleutels WHERE bsn=?{burgerservicenummer} LIMIT 1"
+					includeFieldDefinition="false">
+					<Param name="burgerservicenummer" sessionKey="bsn" />
+				</FixedQuerySender>
+			</SenderPipe>
+
+			<!-- Check if a result was found -->
+			<IfPipe name="HasKey" xpathExpression="count(result/rowset/row) > 0">
+				<Forward name="then" path="StoreKey" />
+				<Forward name="else" path="CreateKey" />
+			</IfPipe>
+
+			<PutInSessionPipe name="StoreKey">
+				<Param name="persoonssleutel" xpathExpression="//field[@name='SLEUTEL']/text()"/>
+				<Forward name="success" path="MapResponse" />
+			</PutInSessionPipe>
+
+			<!-- Create a new key if no key was found -->
+			<HashPipe name="CreateKey" algorithm="SHA" storeResultInSessionKey="persoonssleutel" hashEncoding="Hex" />
```

Comment:
if the goal is to create the persoonssleutel by hashing the bsn, then add getInputFromSessionKey="bsn" here, because right now the input of this pipe is the empty database response